core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/OneOfEnum.java

```diff
@@ -0,0 +1,31 @@
+package com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Target({METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER, TYPE_USE})
+@Retention(RUNTIME)
+@Documented
+@Constraint(validatedBy = OneOfEnumValidator.class)
+public @interface OneOfEnum {
+
+    Class<? extends Enum<?>> enumClass();
+
+    String message() default "Volume Type must be any of {enumClass}";
```

Comment:
if this is a general annotation, then the default message should not contain `Volume Type`

Reply:
done